File: docs/snippets/all/archetypes/clear_simple.py
"""Log and then clear data."""

import dalaran as dl

dl.init("dalaran_example_clear", spawn=True)

vectors = [(1.0, 0.0, 0.0), (0.0, -1.0, 0.0), (-1.0, 0.0, 0.0), (0.0, 1.0, 0.0)]
origins = [
    (-0.5, 0.5, 0.0),
    (0.5, 0.5, 0.0),
    (0.5, -0.5, 0.0),
    (-0.5, -0.5, 0.0),
]
colors = [(200, 0, 0), (0, 200, 0), (0, 0, 200), (200, 0, 200)]

# Log a handful of arrows.
for i, (vector, origin, color) in enumerate(
    zip(vectors, origins, colors, strict=False)
):
    dl.log(
        f"arrows/{i}", dl.Arrows3D(vectors=vector, origins=origin, colors=color)
    )

# Now clear them, one by one on each tick.
for i in range(len(vectors)):
    dl.log(f"arrows/{i}", dl.Clear(recursive=False))  # or `dl.Clear.flat()`

File: docs/snippets/all/archetypes/clear_simple.cpp
// Log and then clear data.

#include <dalaran.hpp>

#include <cmath>
#include <numeric>
#include <string> // to_string
#include <vector>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_clear");
    rec.spawn().exit_on_failure();

    std::vector<dalaran::Vector3D> vectors = {
        {1.0, 0.0, 0.0},
        {0.0, -1.0, 0.0},
        {-1.0, 0.0, 0.0},
        {0.0, 1.0, 0.0},
    };
    std::vector<dalaran::Position3D> origins = {
        {-0.5, 0.5, 0.0},
        {0.5, 0.5, 0.0},
        {0.5, -0.5, 0.0},
        {-0.5, -0.5, 0.0},
    };
    std::vector<dalaran::Color> colors = {
        {200, 0, 0},
        {0, 200, 0},
        {0, 0, 200},
        {200, 0, 200},
    };

    // Log a handful of arrows.
    for (size_t i = 0; i < vectors.size(); ++i) {
        auto entity_path = "arrows/" + std::to_string(i);
        rec.log(
            entity_path,
            dalaran::Arrows3D::from_vectors(vectors[i])
                .with_origins(origins[i])
                .with_colors(colors[i])
        );
    }

    // Now clear them, one by one on each tick.
    for (size_t i = 0; i < vectors.size(); ++i) {
        auto entity_path = "arrows/" + std::to_string(i);
        rec.log(entity_path, dalaran::Clear::FLAT);
    }
}
